packages/nimble-components/src/mapping/icon/tests/mapping-icon.spec.ts

```diff
@@ -0,0 +1,56 @@
+import { html } from '@microsoft/fast-element';
+import { DesignSystem } from '@microsoft/fast-foundation';
+import { MappingIcon, mappingIconTag } from '..';
+import { fixture, type Fixture } from '../../../utilities/tests/fixture';
+import { waitForUpdatesAsync } from '../../../testing/async-helpers';
+import { Icon } from '../../../icon-base';
+
+describe('Icon Mapping', () => {
+    class TestIcon extends Icon {}
+
+    let element: MappingIcon;
+    let connect: () => Promise<void>;
+    let disconnect: () => Promise<void>;
+
+    // prettier-ignore
+    async function setup(): Promise<Fixture<MappingIcon>> {
+        return fixture<MappingIcon>(html`
+        <${mappingIconTag}
+            key="foo"
+            text="foo"
+            icon="nimble-test-icon">
+        </${mappingIconTag}>`);
+    }
+
+    it('should export its tag', () => {
+        expect(mappingIconTag).toBe('nimble-mapping-icon');
+    });
+
+    it('can construct an element instance', () => {
+        expect(document.createElement('nimble-mapping-icon')).toBeInstanceOf(
+            MappingIcon
+        );
+    });
+
+    it('resolves icon after it is defined', async () => {
+        ({ element, connect, disconnect } = await setup());
+        await connect();
+        await waitForUpdatesAsync();
+
+        expect(element.resolvedIcon).toBeUndefined();
+
+        DesignSystem.getOrCreate()
+            .withPrefix('nimble')
+            .register(
+                TestIcon.compose({
+                    baseName: 'test-icon',
+                    template: html`<nimble-xmark-icon></nimble-xmark-icon>`
+                })()
+            );
```

Comment:
Looks like I don't even need to define a `TestIcon` class.

Reply:
I'm not really sure what you mean, but you can't register the same class twice with the different names like the source seems to be doing with the XMark class:
![image](https://github.com/ni/nimble/assets/1588923/d93e6aaa-e97f-44ed-8c55-40581118c28a)

please create a new class or clearly document what's going on and why that's not an issue here because I don't understand it.

Reply:
Maybe compose is creating a new antonymous class or something. I dunno just create a new text class that extends `IconXmark`

Reply:
I'm not sure if `IconXmark` wasn't being defined before I registered it with a new name, or what. Anyway, I'm just using a new class now.